powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/controller/AuthMethodController.java

```diff
@@ -77,17 +80,23 @@
     @Autowired
     private MessageTranslationService messageTranslationService;
 
+    @Autowired
+    private OperationSessionService operationSessionService;
+
+    @Autowired
+    private HttpServletRequest request;
+
     protected GetOperationDetailResponse getOperation() throws AuthStepException {
         final UserOperationAuthentication pendingUserAuthentication = authenticationManagementService.getPendingUserAuthentication();
         if (pendingUserAuthentication != null) {
             String operationId = pendingUserAuthentication.getOperationId();
             if (operationId != null) {
                 return getOperation(operationId);
             } else {
-                return null;
+                throw new AuthStepException("operation.notAvailable", new NullPointerException());
```

Comment:
This looks weird - do not use `new NullPointerException()`, just make sure to log the warning and provide appropriate `message` for the exception. Currently, we are incorrectly "hacking" `message` to pass a localization key - this should be used for providing readable information in the exception.  We can either introduce new exception subclass `class OperationNotAvailableException extends AuthStepException`, or extend `AuthStepException` by adding `code` field with extended info about error that caused this exception to be raised.

Reply:
... we have it more times... I will open separate issue on this.

Reply:
You are right, we should do the error handling more properly, this is a misuse of the AuthStepException class.